ts-rs/src/export.rs

```diff
@@ -36,6 +99,11 @@ pub(crate) fn export_type<T: TS + ?Sized + 'static>() -> Result<(), ExportError>
 pub(crate) fn export_type_to<T: TS + ?Sized + 'static, P: AsRef<Path>>(
     path: P,
 ) -> Result<(), ExportError> {
+    // Lock to make sure only one file will be written at a time.
+    // In the future, it might make sense to replace this with something more clever to only prevent
+    // two threads from writing the **same** file concurrently.
+    static FILE_LOCK: Mutex<()> = Mutex::new(());
+
```

Comment:
@escritorio-gustavo Regarding the lock we introduced here:

I've experimented with replacing this with a `static EXPORTED: Mutex<Option<HashSet<PathBuf>>>` to track which files were written, and to skip an export if the file was already written to.

The case in our test suite where this is most helpfull is `generics.rs`, where 30 distinct types are being exported while 37 unnecesarry exports are being skipped.

That being said, I don't think we actually need to do that. `generics.rs` ran in 50ms before, and runs in 32ms after this (though with much higher variance). That seems plenty fast.

Reply:
It definitely seems fast enough as is, but if we can reduce the amount of fs writes without any negative impact, I think it's a good idea

Reply:
Maybe push that into #295 as a part of the test performance improvements?